backend/index.js

```diff
@@ -0,0 +1,65 @@
+const express = require('express')
+const app = express()
+const mongoose = require('mongoose')
+const Recipe = require('./models/recipeSchema')
+const connection_url = "mongodb+srv://dbMeganTrieu:YHb5aXyiQPomvX4Q@cluster0.rtwvj.mongodb.net/RecipesDB?retryWrites=true&w=majority"
+
+mongoose.connect(connection_url)
+    .then(() => console.log('Successfully connected'))
+    .catch((error) => console.error(`Could not connect due to ${error}`))
+    
+app.use(express.json())
+app.get('/', (req, res) => {
+  res.send('Hello world!')
+})
+app.listen(3001)
+
+
+app.get("/api/recipe", async (req, res) => {
+    const recipes = await Recipe.find({})
+    res.send(recipes)
+})
+
+app.get("/api/recipe/:name", async (req, res) => {
+    const recipeName = req.params.recipeName;
+    const recipe = await Recipe.findOne({ 
+        recipeName: recipeName 
+    })
+    res.send(recipe)
+})
+
+app.post("/api/recipe", async (req, res) => {
+    const {name, link, desc, recipeImage, size, ingredients, prepTime, prep} = req.body;
+    let recipe = new Recipe({name, link, desc, recipeImage, size, ingredients, prepTime, prep})
+    try{
+        recipe = await recipe.save()
+        res.send('sucessfully added recipe')
+    }
+    catch(error){
+        res.status(500).send(error)
```

Comment:
rather than using 500, a generic error code, we can use error.code to send a more specific/helpful error! Similarly, we can send(error.message) to display more helpful information.